Backend/ServiceLayer/Services/ResetService.cs

```diff
@@ -0,0 +1,229 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+using DataAccessLayer.Database;
+using DataAccessLayer.Models;
+using DataAccessLayer.Repositories;
+using MimeKit;
+using MailKit;
+using MailKit.Net.Smtp;
+
+
+namespace ServiceLayer.Services
+{
+    public class ResetService
+    {
+
+        #region Constants for the reset service
+        //Variable for how long the token is supposed to be live
+        private const double TimeToExpire = 5;
+        //Constant for the password reset email subject line
+        private const string resetPasswordSubjectString = "KFC SSO Password Reset";
+        //URL to append the reset ID to. Sent to the user in the email body.
+        private const string resetControllerURL = "kfcsso.com/api/reset/resetpassword/?id=";
```

Comment:
This shouldn't be hardcoded. What if we change domains? Or another part of the stack changes?

This URL should be derived from the request headers coming in.

Perhaps leave off the domain part and build the rest like you are, using the heads to figure out that first part.